src/server/middleware/twitter.middleware.ts

```diff
@@ -133,30 +133,47 @@ export async function hideTwitterReplies(
 }
 
 async function blockUsers(
-  apiCredentials: TwitterApiCredentials,
+  credentials: TwitterApiCredentials,
   request: BlockTwitterUsersRequest
 ): Promise<BlockTwitterUsersResponse> {
-  const client = createAxiosInstance(apiCredentials, request.credential);
-  const requestUrl = 'https://api.twitter.com/1.1/blocks/create.json';
+  const client = createAxiosInstance(credentials, request.credential);
   const response: BlockTwitterUsersResponse = {};
+  const id = getUserIdFromCredential(request.credential);
+  if (!id) {
+    response.error = 'Missing Twitter user ID in access token';
+    return response;
+  }
+  let quotaExhaustedErrors = 0;
+  let otherErrors = 0;
   const requests = request.users.map((user) =>
     client
       .post<BlockTwitterUsersResponse>(
-        requestUrl,
-        {},
-        { params: { screen_name: user } }
+        `https://api.twitter.com/2/users/${id}/blocking`,
+        { target_user_id: user.id_str }
       )
-      .catch((e) => {
-        console.error(`Unable to block Twitter user: @${user} because ${e}`);
+      .catch((e: AxiosError) => {
+        if (
+          e.response?.status === 429 ||
```

Comment:
Could we use a constant for the value of 429? Maybe from https://www.npmjs.com/package/http-status-codes?

Reply:
Good call! Looks like Angular has an [HttpStatusCode](https://angular.io/api/common/http/HttpStatusCode) enum we can use. Updated.